app/scripts/lib/stores/persistence-manager.test.ts

```diff
@@ -0,0 +1,163 @@
+      await expect(manager.set({ appState: { test: true } })).rejects.toThrow(
+        'MetaMask - metadata must be set before calling "set"',
+      );
+    });
+
+    it('calls localStore.set with the correct arguments once metadata is set', async () => {
+      manager.setMetadata({ version: 10 });
+
+      await manager.set({ appState: { test: true } });
+
+      expect(mockStoreSet).toHaveBeenCalledTimes(1);
+      expect(mockStoreSet).toHaveBeenCalledWith({
+        data: { appState: { test: true } },
+        meta: { version: 10 },
+      });
+    });
+
+    it('logs error and captures exception if store.set throws', async () => {
+      manager.setMetadata({ version: 10 });
+
+      const error = new Error('store.set error');
+      mockStoreSet.mockRejectedValueOnce(error);
+
+      await manager.set({ appState: { broken: true } });
+      expect(captureException).toHaveBeenCalledWith(error);
+      expect(log.error).toHaveBeenCalledWith(
+        'error setting state in local store:',
+        error,
+      );
+    });
+
+    it('captures exception only once if store.set is called and throws multiple times', async () => {
+      manager.setMetadata({ version: 10 });
+
+      const error = new Error('store.set error');
+      mockStoreSet.mockRejectedValue(error);
+
+      await manager.set({ appState: { broken: true } });
+      await manager.set({ appState: { broken: true } });
+
+      expect(captureException).toHaveBeenCalledTimes(1);
+    });
+
+    it('captures exception twice if store.set fails, then succeeds and then fails again', async () => {
+      manager.setMetadata({ version: 17 });
+
+      const error = new Error('store.set error');
+      mockStoreSet.mockRejectedValueOnce(error);
+
+      await manager.set({ appState: { broken: true } });
+
+      mockStoreSet.mockReturnValueOnce({
+        data: { appState: { broken: true } },
+      });
+      await manager.set({ appState: { broken: true } });
+
+      mockStoreSet.mockRejectedValueOnce(error);
+
+      await manager.set({ appState: { broken: true } });
+
+      expect(captureException).toHaveBeenCalledTimes(2);
+    });
+  });
+
+  describe('get', () => {
+    it('returns undefined and clears mostRecentRetrievedState if store returns empty', async () => {
+      mockStoreGet.mockReturnValueOnce({});
+      const result = await manager.get();
+      expect(result).toBeUndefined();
+      expect(manager.mostRecentRetrievedState).toBeNull();
+    });
+
+    it('returns undefined if store returns null', async () => {
+      mockStoreGet.mockReturnValueOnce(null);
+      const result = await manager.get();
+      expect(result).toBeUndefined();
+      expect(manager.mostRecentRetrievedState).toBeNull();
+    });
+
+    it('updates mostRecentRetrievedState if extension has not been initialized', async () => {
+      mockStoreGet.mockResolvedValueOnce({ data: { config: { foo: 'bar' } } });
+
+      await manager.get();
```

Comment:
Can we also assert the result of `get` is `data: { config: { foo: 'bar' } }`?

Reply:
done in 1d4aa025db